core/src/plugins/filed/python/python-fd.cc

```diff
@@ -138,7 +138,7 @@ static PyThreadState* mainThreadState{nullptr};
 extern "C" {
```

Comment:
I just stumbled over this and it is at least a bit weird:
- why would you make `PyVersion()` have C linkage?
- why would we have `#ifdef __cplusplus` in a TU that is always built with a C++ compiler?
- why do we have functions defined with C linkage, when they were forward-declared with C++ linkage? I would have understood it the other way around though...

Reply:
I think I figured it out:
`static` overrides `extern "C"`. Thus `extern "C"` has no effect on the static functions which have internal linkage only.

Thus, the only functions that require `extern "C"` are `loadPlugin()` and `unloadPlugin()` as these will be looked up by name using `dlsym()`.
I guess it is sensible to reduce the extern "C" scope to only include the functions where it is actually required.

Reply:
removed the `__cplusplus` stuff since that did not really make sense.  Moved the PyVersion function up so its not part of the extern "C" block anymore.  I also moved the rest of the code around a bit so that now everything static is instead inside an anonymous namespace.

Reply:
> I think I figured it out: `static` overrides `extern "C"`. Thus `extern "C"` has no effect on the static functions which have internal linkage only.
> 
> Thus, the only functions that require `extern "C"` are `loadPlugin()` and `unloadPlugin()` as these will be looked up by name using `dlsym()`. I guess it is sensible to reduce the extern "C" scope to only include the functions where it is actually required.

Werent `loadPlugin()` and `unloadPlugin()` the ony functions declared `extern "C"` anyways (apart from `PyVersion` of course) ?

Reply:
effectively yes, but with the `static` functions in the `extern "C"` block it looked like they had C-linkage - unless you realize that `static` means no linkage.
So it wasn't wrong, but (at least for me) a bit confusing.